src/components/Amalgamation/AmalgamatingBusinesses.vue

```diff
@@ -165,80 +177,66 @@ export default class AmalgamatingBusinesses extends Mixins(CommonMixin) {
   // Local properties
   initialBusinessLookupObject = EmptyBusinessLookup
   businessTableValid = false
+  snackbar = false
+  snackbarText = ''
 
   // Button properties
   isAddingAmalgamatingBusiness = false
   isAddingAmalgamatingForeignBusiness = false
 
-  readonly BusinessLookupServices = BusinessLookupServices
-
-  // Cancel button in "Add an Amalgamating Business" is pressed.
-  addAmalgamatingBusinessCancel (): void {
-    this.isAddingAmalgamatingBusiness = false
-    this.setAmalgamatingBusinessesValid(true)
-  }
-
-  // "Add an Amalgamating Business" button is pressed.
-  onAddBusinessClick (): void {
-    this.isAddingAmalgamatingBusiness = true
-    this.isAddingAmalgamatingForeignBusiness = false
-    this.setAmalgamatingBusinessesValid(false)
-  }
-
-  // "Add an Amalgamating Foreign Business" button is pressed.
-  onAddForeignBusinessClick (): void {
-    this.isAddingAmalgamatingBusiness = false
-    this.isAddingAmalgamatingForeignBusiness = true
-    this.setAmalgamatingBusinessesValid(false)
-  }
-
   async saveAmalgamatingBusiness (businessLookup: BusinessLookupIF): Promise<void> {
-    let business = null
-
-    // Get the amalgamating business information, mailing address, and email if in LEAR.
+    // Get the amalgamating business information, auth info and addresses, and email if in LEAR.
     // Otherwise, return the businesslookup object.
     const data = await Promise.all([
       LegalServices.fetchBusinessInfo(businessLookup.identifier),
       AuthServices.fetchAuthInfo(businessLookup.identifier),
-      LegalServices.fetchAddresses(businessLookup.identifier)
-    ]).catch((error) => {
-      return error
-    })
+      LegalServices.fetchAddresses(businessLookup.identifier),
+      LegalServices.fetchFilings(businessLookup.identifier)
```

Comment:
This is WIP -- there's an outstanding question whether we need to fetch the filings after all.

Reply:
Yes, we need to fetch the latest filing to see if it's future-effective.

We also need to fetch the state filing, if it's declared, to see if business is in limited restoration.

In a way, it would be nice to not fetch one of these if the other is true. On the other hand, the table will be validated over and over and I wouldn't want to fetch db data every time.